kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -155,9 +155,9 @@ public Transaction build(Engine engine) {
       if (!newWriterFeatures.isEmpty()) {
         logger.info("Automatically enabling writer features: {}", newWriterFeatures);
         shouldUpdateProtocol = true;
-        List<String> oldWriterFeatures = protocol.getWriterFeatures();
+        Set<String> oldWriterFeatures = protocol.getWriterFeatures();
```

Comment:
Not a comment, but a question on the existing code for my understanding: TableFeatures.extractAutomaticallyEnabledWriterFeatures looks for `writer` features. So what is the mechanism to enable reader-writer features such as `timestampNtz`?